keras/metrics.py

```diff
@@ -71,6 +71,25 @@ def cosine_proximity(y_true, y_pred):
     return -K.mean(y_true * y_pred)
 
 
+def matthews(y_true, y_pred):
+    ''' Matthews correlation coefficient
+    '''
+    y_pred_pos = K.round(y_pred)
+    y_pred_neg = 1 - y_pred_pos
+    
+    y_pos = y_true
+    y_neg = 1 - y_true
+
+    TP = K.sum(y_pos * y_pred_pos)
+    TN = K.sum(y_neg * y_pred_neg)
+
+    FP = K.sum(1 - y_neg * y_pred_pos)
+    FN = K.sum(1 - y_pos * y_pred_neg)
+
+    MCC = (TP*TN-FP*FN)/K.sqrt((TP+FP)*(TP+FN)*(TN+FP)*(TN+FN))
```

Comment:
So, this can be NaN if either all your predictions or all of your ground truth are from a single class, though that is there by definition of the metric, is that an issue? Could you give me some pointers to the sort of fuzzing you want done? I'm not super familiar with numerical issues.